### ISSUE TITLE
[Bug Report] hf_generate right-pads batched strings and drops the attention mask

### ISSUE THREAD

**Issue by emerardd** [TransformerLensOrg/TransformerLens#1760]
**Describe the bug**

`TransformerBridge.hf_generate()` silently produces incorrect continuations for shorter rows in a
variable-length `list[str]` batch.

The list-input branch tokenizes with `padding=True` using the tokenizer's current padding side. For
GPT-2-family tokenizers this is normally right padding. Although the tokenizer returns an
`attention_mask`, `hf_generate()` extracts only `input_ids` and calls
`original_model.generate(input_ids, **generation_kwargs)`, so Hugging Face sees the pad columns as
real context.

This is difficult to notice because the pad token is commonly the EOS token and
`skip_special_tokens=True` hides the padding in the returned text. The call succeeds and returns a
plausible but different continuation from generating the same prompt alone.

**Code example**

On current `dev` (`7edc5ddf`), using `distilgpt2`, greedy generation, and two deliberately unequal
prompts:

```python
prompts = ["The capital of France is", "Hi"]

batched = bridge.hf_generate(
    prompts,
    max_new_tokens=8,
    do_sample=False,
    return_type="input",
)
solo = [
    bridge.hf_generate(
        prompt,
        max_new_tokens=8,
        do_sample=False,
        return_type="input",
    )
    for prompt in prompts
]
```

Observed output for the short row:

```text
batched: 'HiThe following is a list of the most'
solo:    'Hi everyone,\n\n\n\n\n\n'
```

`TransformerBridge.generate(prompts, ...)` agrees with the solo control for the short row because
that path forces left padding and preserves the relevant mask semantics.

The current equal-length batch test uses `['Hello', 'World']`, so no padding is emitted and it cannot
detect this failure.

**Impact**

- Every row shorter than the longest prompt can be continued from padding rather than its last real
  token.
- Batched `hf_generate()` can disagree with row-by-row `hf_generate()` under greedy decoding.
- Stateful/SSM `generate(use_past_kv_cache=False)` delegates to `hf_generate()`, so its batched string
  fallback reaches the same path.

**Suggested scope**

For `list[str]` input, force left padding without permanently mutating the shared tokenizer, retain
the tokenizer's `attention_mask`, and pass it to `original_model.generate()`. Preserve tensor-input
behavior and HF `ModelOutput` flags.

A larger alternative would route through `to_tokens()`, but that also changes BOS and truncation
semantics. I plan to keep this fix focused unless maintainers prefer that broader unification.

**System Info**

- Installed from source with `uv`
- Windows 11 / PowerShell 7
- Python 3.12.10
- Reproduced on `upstream/dev` commit `7edc5ddfccc24619109e3c98db29a870d99977ec`

**Additional context**

Issue #1690 / PR #1701 fixed related batched padding behavior in the normal Bridge tokenization and
generation paths, but did not change `hf_generate()`. This report is limited to the HF-delegation
path.

### Checklist

- [x] I have checked that there is no similar issue in the repo
